flaskr/backservice.py

```diff
@@ -0,0 +1,181 @@
+
+        return query
+    
+    if(request_type == "POST"):
+        # INSERT INTO TABLE_NAME(id, first_name, last_name, email, job_id) VALUES (%s, %s, %s, %s, %s)
+        if action == "create":
+            values = []
+            columns = []
+            for key, value in record.items():
+                columns.append(key)
+                values.append(f"\'{value}\'")
+            
+            query = "INSERT INTO {} ({}) VALUES ({})".format(TABLE_NAME, ", ".join(columns), ", ".join(values))
+
+        if(action == "update"):
+            key_value_pairs = ", ".join(["{}='{}'".format(key, value) for key, value in record.items()])            
+            query = f"UPDATE {TABLE_NAME} SET {key_value_pairs} WHERE {KEY_COLUMNS_NAMES}='{record.get(KEY_COLUMNS_NAMES)}'"
+        
+        return query
+
+
+@app.route("/candidate", methods=["GET", "POST", "DELETE"])
+def apiHandelCandidate():
+    GET_ALL_CANDIDATES = f"SELECT * FROM {TABLE_NAME} "
+    # TODO: CHANGE THE 100 TO A VARIABLE FROM USERS WITH DEFAULT OF 100
+    
+    messages = {
+        "get": {
+            "success": "Successful in fetching all candidates",
+            "error": "Candidates data for GET request not found"
+        },
+        "create": {
+            "success": "Successful in creating all new candidates",
+            "error": "Candidates data for POST (create) request not found"
+        },
+        "update": {
+            "success": "Successful in updating all candidates",
+            "error": "Candidates data for POST (update) request not found"
+        },
+        "delete": {
+            "success": "Successful in deleting all candidates",
+            "error": "Candidates data for DELETE request not found"
+        }
+    }
+    query = ""
+    success = ""
+    error = ""
+
+    try:
+        # Store the url query params
+        args = request.args.to_dict() 
+        
+        # Check connectivity to DB
+        getReady = apiHandelReady()
+        if(getReady.status_code != 200):
+            return getReady
+        
+        connection = psycopg2.connect(host=DB_HOST, port=DB_PORT, dbname=DB_NAME, user=DB_USERNAME, password=DB_PASSWORD, options = dbConnectionOptions)
+        cur = connection.cursor()
+        
+        if(request.method == "GET"):  
+            success = messages.get("get", {}).get("success")         
+            error =  messages.get("get", {}).get("error")
+
+            if(args == {}): # Get all candidates               
+                return helper.getResponseFromDB(connection, cur, GET_ALL_CANDIDATES + RECORDS_LIMIT, success, error, request.method)
+            else: # Get one/many candidates
+                query = createQueryByRequestType(args, request.method)
+                return helper.getResponseFromDB(connection, cur, f"{query} {RECORDS_LIMIT}", success, error, request.method)
+                
+        if(request.method == "POST"):
+            action = args.pop("action") # action=update or action=create
```

Comment:
It is a convinent way to:
1. Use the value in the check create/update here
2. Remove the vaule since I do not need it any more (in argsToSetOfValues it is not needed and could create incorrect query)
3. Prepare the args variable argsToSetOfValues (the sets of key=value creation)